pkg/reconciler/taskrun/taskrun.go

```diff
@@ -598,3 +605,21 @@ func getLabelSelector(tr *v1alpha1.TaskRun) string {
 	}
 	return strings.Join(labels, ",")
 }
+
+// update SidecarStatuses based on Pod ContainerStatuses when Sidecars
+// have been stopped.
+func updateSidecarStatus(pod *corev1.Pod, tr *v1alpha1.TaskRun, c *Reconciler) error {
```

Comment:
Instead of using current State (`s.State`) from ContainerStatus on Pod with stopped Sidecars, what could be done is to create a Terminated Status with the following information:

```go
corev1.ContainerState{
    Terminated: &corev1.ContainerStateTerminated{
        ExitCode: 0,
	Reason:   "Completed",
        Message:  "Sidecar container successfully stopped by nop-image",
        StartedAt: <Use value from LastTerminatedState instead of current State (s.LastTerminatedState.Terminated.StartedAt)>,
        FinishedAt: <Use value from LastTerminatedState instead of current State (s.LastTerminatedState.Terminated.FinishedAt)>,
        ContainerID: <Use value from LastTerminatedState instead of current State (s.LastTerminatedState.Terminated.ContainerID)>,
}
```

Using the LastTerminationStatus, if exit code == 137, it can be assumed that the sidecar was killed.

Using the LastTerminationStatus would also provide more accurate information about the Start/End time of the Sidecar as well as information about the actual image that was used for the Sidecar.